storm-client/src/jvm/org/apache/storm/daemon/Task.java

```diff
@@ -252,6 +252,8 @@ private TopologyContext mkTopologyContext(StormTopology topology) throws IOExcep
             executor.getIntervalToTaskToMetricToRegistry(),
             executor.getOpenOrPrepareWasCalled(),
             workerData.getMetricRegistry());
+        executor.setMetricRegistry(workerData.getMetricRegistry());
```

Comment:
Should we put this line of code outside this function? They don't seem to be related

Reply:
This should happen on Executor level instead of Task level.

Reply:
Yes, I agree.  Will move.